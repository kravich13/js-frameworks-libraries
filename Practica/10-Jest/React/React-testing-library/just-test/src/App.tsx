import React, { useEffect, useState } from 'react'
import './App.css'
import { List } from './components/List'

const getUser = (): Promise<string> => Promise.resolve('Max')

const App: React.FC = () => {
  const [pState, setPState] = useState<boolean>(false)
  const [user, setUser] = useState<string>('')

  useEffect((): void => {
    const loadUser = async (): Promise<void> => {
      try {
        const userData = await getUser()
        setUser(userData)
      } catch (err) {
        return
      }
    }
    loadUser()
  }, [setUser])

  return (
    <div className="App">
      <div className="sync-test">
        <h3>Hello</h3>
        <p>Какая-то инфа</p>
        <p>{pState ? 'Max' : 'Vlad'} Kravich</p>
        <img alt="нет картинки"></img>
        <label htmlFor="search" />
        <input type="text" placeholder="Текст..." defaultValue="13" />
        <button
          data-testid="buttonApp-click"
          onClick={(): void => setPState(true)}
        >
          Нажми
        </button>
      </div>

      <div className="async-test">{user && <h2>Logged in as {user}</h2>}</div>

      <div className="approval">
        <p className="classTest">Привет</p>
        <p id="kravich1">Ку</p>
      </div>

      <List />
    </div>
  )
}

export default App
