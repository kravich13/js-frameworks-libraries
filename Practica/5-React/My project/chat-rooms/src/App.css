.App {
  font-family: Arial, Helvetica, sans-serif;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
}

header {
  width: 100%;
}
section {
  display: flex;
  width: 1210px;
  height: 800px;
  border: 1px solid lightcoral;
}

.done {
  list-style-type: none;
  background: rgba(240, 240, 240, 0.8);
  color: black;
  padding: 5px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.done1 {
  list-style-type: none;
  background: rgb(60, 223, 201);
  color: black;
  padding: 5px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
.done2 {
  list-style-type: none;
  background: rgba(60, 223, 201, 0.5);
  color: black;
  padding: 5px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
