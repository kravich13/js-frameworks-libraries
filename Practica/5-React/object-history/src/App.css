.App {
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
}

.max {
  color: greenyellow;
}
.vlad {
  color: red;
}
.katya {
  color: pink;
}

.max,
.vlad,
.katya {
  width: 30%;
  background: rgb(58, 58, 58);
  font-weight: 600;
  border-radius: 13px;
}

ul {
  margin: 10px 0;
  padding: 0;
  align-items: center;
  list-style: none;
}
