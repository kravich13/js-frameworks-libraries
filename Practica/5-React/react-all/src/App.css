.testLi {
    background: rgb(82, 82, 82);
    color: white;
    margin: 5px 0;
    padding: 5px 7px;
    border: 1px solid black;
}
.clickLi {
    color: rgb(187, 11, 11);
    background: white;
}

#root {
    padding: 20px;
}

.App {
    color: rgb(202, 0, 0);
    background: lightgray;
    width: auto;
    /* font-weight: 600; */
    border-radius: 5px;
    border: 1px solid rgb(170, 0, 0);
    padding: 10px;
}