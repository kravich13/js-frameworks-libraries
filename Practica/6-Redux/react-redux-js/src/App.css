.App {
  text-align: center;
}

.form {
  text-align: start;
  margin-left: 10%;
}
.form form {
  display: flex;
  flex-direction: column;
}
.form form input {
  width: 400px;
}
.form form button {
  width: 170px;
}
.posts {
  display: flex;
  justify-content: space-around;
}
.card {
  border: 1px solid rgb(59, 59, 59);
  width: 500px;
}
