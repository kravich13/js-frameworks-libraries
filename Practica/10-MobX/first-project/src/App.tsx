import React from 'react';
import './App.css';
import { AsyncPosts } from './components/AsyncPosts';
import { PostForm } from './components/PostForm';
import { Posts } from './components/Posts';

function App() {
  return (
    <div className="App">
      <PostForm />
      <Posts />
      <AsyncPosts />
    </div>
  );
}

export default App;
