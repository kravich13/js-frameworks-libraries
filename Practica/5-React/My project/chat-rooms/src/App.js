import React, { useEffect, useRef, useState } from 'react'
import './App.css'
import Header from './all-blocks/Header'
import Rooms from './all-blocks/Rooms'
import WindowChat from './all-blocks/WindowChat'
import Context from './context'

function App() {
  const [value, setValue] = useState('initial')
  const countRender = useRef(1)
  const inputRef = useRef(null)

  useEffect(() => {
    countRender.current++

    // Это сам DOM элемент, со всеми его доступными свойствами
    console.log(inputRef.current.value)
  })

  // При клике на кнопку будет фокус на элементе inputRef
  const focus = () => inputRef.current.focus()

  function clickRoom(id) {
    // console.log('нажал в другом компоненте')
  }

  return (
    <div className="App">
      <p>Значение: {countRender.current}</p>

      <input
        ref={inputRef}
        type="text"
        onChange={(event) => setValue(event.target.value)}
        value={value}
      />

      <button onClick={focus}>Фокус</button>

      <Header />

      <Context.Provider value={{ clickRoom: clickRoom }}>
        <section>
          <Rooms />
          <WindowChat />
        </section>
      </Context.Provider>
    </div>
  )
}

export default App
