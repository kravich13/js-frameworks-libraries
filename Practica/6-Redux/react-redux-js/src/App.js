import './App.css'
import PostForm from './components/PostForm'
import Posts from './components/Posts'
import FetchedPosts from './components/FetchedPosts'

function App() {
  return (
    <div className="App">
      <div className="App">
        <div className="form">
          <h1>Форма отправки постов</h1>
          <PostForm />
        </div>
        <div className="posts">
          <div>
            <h1>Синхронные посты</h1>
            <Posts />
          </div>
          <div>
            <h1>Асинхронные посты</h1>
            <FetchedPosts posts={[]} />
          </div>
        </div>
      </div>
    </div>
  )
}

export default App
