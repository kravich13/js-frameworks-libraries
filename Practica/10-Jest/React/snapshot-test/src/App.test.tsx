import renderer from 'react-test-renderer'
import { App } from './App'

describe('App other', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
