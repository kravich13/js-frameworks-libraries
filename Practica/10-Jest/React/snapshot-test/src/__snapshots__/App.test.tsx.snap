// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App other renders correctly 1`] = `
<div
  className="App"
>
  <ol>
    <li>
      Vlad
    </li>
    <li>
      Max
    </li>
    <li>
      Kravich
    </li>
  </ol>
</div>
`;
